docs/pfcwd/README_PFCWD_Various_Reboot_Scenarios.md

```diff
@@ -0,0 +1,100 @@
+
+### PFC storm detection
+
+The PFC storm detection is for a switch to detect a lossless queue is receiving PFC storm from its link partner and the queue is in a paused state over _T0_ amount of time. Even when the queue is empty, as soon as the duration for a queue in paused state exceeds T0 amount of time, the watchdog should detect such storm. T0 is called _PFC storm detection time_.
+
+### PFC storm mitigation
+
+Once PFC storm is detected on a queue, the watchdog can then have two actions, drop and forward at per queue level. When drop action is selected, following actions need to be implemented.
+
+* All existing packets in the output queue are discarded
+* All subsequent packets destine to the output queue are discarded
+* All subsequent packets received by the corresponding priority group of this queue are discarded including the pause frames received. As a result, the switch should not generate any pause frame to its neighbor due to congestion of this output queue.
+
+When forward action is selected, following actions need to be implemented.
+
+* The queue no longer honor the PFC frames received. All packets destined to the queue are forwarded as well as those packets that were in the queue.
+
+### PFC storm restoration
+
+The watchdog should continue count the PFC frames received on the queue. If there is no PFC frame received over T1 period. Then, re-enable the PFC on the queue and stop dropping packets if the previous mitigation was drop. T1 is called _PFC storm restoration time_.
+
+### PFC watchdog implementation
+
+PFC watchdog polls the states of each lossless queue every T2 period. T2 is called _polling interval_. To reduce CPU overhead, T2 is typically of hundreds of milliseconds.
+
+## PFC watchdog commands on SONiC
+
+To get PFC watchdog configuration:
+
+<code> $ pfcwd show config</code>
+
+To get PFC watchdog statistics:
+
+<code>$ pfcwd show stats</code>
+
+To start PFC watchdog
+
+<code>$ sudo pfcwd start --action drop ports all detection-time [detection time in ms] --restoration-time [restoration time in ms]</code>
+
+To stop PF C watchdog
+
+<code>$ sudo pfcwd stop</code>
+
+## Testbed Setup
+
+The testbed consists of two Keysight ports and a SONiC device under test (DUT) as follows. All the KEYSIGHT ports should have the same bandwidth capacity.
+
+
+``` 
+                            _________
+                           |         |
+KEYSIGHT port 1 ------- et1|   DUT   |et2------ KEYSIGHT port 2
+                           |_________|
+                               et3
+                                |
+                                |
+                        KEYSIGHT port 3
+                        
+```
+## Experiment Steps
+
+1. Do a warm reboot of the SONiC switch:
+
+<code>$sudo warm-reboot [options] </code>
+
+2. Wait for the DUT to come up
+3. Execute the test &quot;PFCWD\_Basic\_All-to-all&quot; test for only one priority (e.g., 3)
```

Comment:
Explain what is the pfcwd basic all to all test

Reply:
This test ( pfcwd basic all to all test ) is part of PR ( https://github.com/Azure/sonic-mgmt/pull/2205 ) - Test Plan : PFC Watchdog test plan and Test Case : 1.3.1 Test Case #1 - PFCWD two senders two receivers.. Overall once this is coded , this will also be added in test plan document : https://github.com/Azure/sonic-mgmt/pull/2205

Reply:
Please provide a link to the testcase in this document

Reply:
PFC WD Test Plan is yet to merge in sonic-mgmt/master - is it ok to provide the link of PR ( #2205 ) in actual document ?